### ISSUE TITLE
Implement softmax loss function

### ISSUE THREAD

**Issue by AditMeh** [AditMeh/deep-learning-from-scratch#3]
In order to work with multiclass classification, I need to implement softmax to create predictions using the output of the final activation layer of size _(n, 1)_.
